include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -554,148 +569,200 @@ namespace internal
                     const bool               integrate_grad);
   };
 
-  template <int dim, int fe_degree, int n_q_points_1d, int n_components, typename Number>
+  template <int dim, int fe_degree, int n_components, typename Number>
   inline
   void
-  FEEvaluationImpl<MatrixFreeFunctions::tensor_gausslobatto, dim,
-                   fe_degree, n_q_points_1d, n_components, Number>
-                   ::evaluate (const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
-                               VectorizedArray<Number> *values_dofs[],
-                               VectorizedArray<Number> *values_quad[],
-                               VectorizedArray<Number> *gradients_quad[][dim],
-                               VectorizedArray<Number> *hessians_quad[][(dim*(dim+1))/2],
-                               VectorizedArray<Number> *scratch_data,
-                               const bool               evaluate_val,
-                               const bool               evaluate_grad,
-                               const bool               evaluate_lapl)
+  FEEvaluationImplTransformSpectral<dim, fe_degree, n_components, Number>
+  ::evaluate (const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
+              VectorizedArray<Number> *values_dofs[],
+              VectorizedArray<Number> *values_quad[],
+              VectorizedArray<Number> *gradients_quad[][dim],
+              VectorizedArray<Number> *hessians_quad[][(dim*(dim+1))/2],
+              VectorizedArray<Number> *,
+              const bool               ,
+              const bool               evaluate_grad,
+              const bool               evaluate_lapl)
   {
     typedef EvaluatorTensorProduct<evaluate_evenodd, dim, fe_degree, fe_degree+1,
             VectorizedArray<Number> > Eval;
-    Eval eval (shape_info.shape_val_evenodd,
-               shape_info.shape_gra_evenodd,
-               shape_info.shape_hes_evenodd,
-               shape_info.fe_degree,
-               shape_info.n_q_points_1d);
-
-    // These avoid compiler errors; they are only used in sensible context but
+    Eval eval_val (shape_info.shape_val_evenodd,
+                   shape_info.shape_gra_evenodd,
+                   shape_info.shape_hes_evenodd,
+                   shape_info.fe_degree,
+                   shape_info.n_q_points_1d);
+    Eval eval(shape_info.shape_values,
```

Comment:
The shape functions underlying `shape_grad_spectral_eo` refer to a different polynomial space than `shape_val_evenodd`.

And indeed, the names in this part are horrible. Since they are in the internal namespace, I will simply go along and make them uniform:
```
shape_values
shape_values_eo
shape_gradients
shape_gradients_eo
shape_hessians
shape_hessians_eo
shape_gradients_collocation_eo
shape_hessians_collocation_eo
```
(the latter name depends on whether we find spectral or collocation more intuitive; the more I think about it the more I like collocation).